avm/res/network/application-gateway/main.bicep

```diff
@@ -0,0 +1,608 @@
+            maxCapacity: autoscaleMaxCapacity
+            minCapacity: autoscaleMinCapacity
+          }
+        : null
+      backendAddressPools: backendAddressPools
+      backendHttpSettingsCollection: backendHttpSettingsCollection
+      backendSettingsCollection: backendSettingsCollection
+      customErrorConfigurations: customErrorConfigurations
+      enableHttp2: enableHttp2
+      firewallPolicy: !empty(firewallPolicyId)
+        ? {
+            id: firewallPolicyId
+          }
+        : null
+      forceFirewallPolicyAssociation: !empty(firewallPolicyId)
+      frontendIPConfigurations: frontendIPConfigurations
+      frontendPorts: frontendPorts
+      gatewayIPConfigurations: gatewayIPConfigurations
+      globalConfiguration: endsWith(sku, 'v2')
+        ? {
+            enableRequestBuffering: enableRequestBuffering
+            enableResponseBuffering: enableResponseBuffering
+          }
+        : null
+      httpListeners: httpListeners
+      loadDistributionPolicies: loadDistributionPolicies
+      listeners: listeners
+      privateLinkConfigurations: privateLinkConfigurations
+      probes: probes
+      redirectConfigurations: redirectConfigurations
+      requestRoutingRules: requestRoutingRules
+      routingRules: routingRules
+      rewriteRuleSets: rewriteRuleSets
+      sku: {
+        name: sku
+        tier: endsWith(sku, 'v2') ? sku : substring(sku, 0, indexOf(sku, '_'))
+        capacity: autoscaleMaxCapacity > 0 && autoscaleMinCapacity >= 0 ? null : capacity
+      }
+      sslCertificates: sslCertificates
+      sslPolicy: sslPolicyType != 'Predefined'
+        ? {
+            cipherSuites: sslPolicyCipherSuites
+            minProtocolVersion: sslPolicyMinProtocolVersion
+            policyName: empty(sslPolicyName) ? null : sslPolicyName
+            policyType: sslPolicyType
+          }
+        : {
+            policyName: empty(sslPolicyName) ? null : sslPolicyName
+            policyType: sslPolicyType
+          }
+      sslProfiles: sslProfiles
+      trustedClientCertificates: trustedClientCertificates
+      trustedRootCertificates: trustedRootCertificates
+      urlPathMaps: urlPathMaps
+    },
+    (enableFips
+      ? {
+          enableFips: enableFips
+        }
+      : {}),
+    (!empty(webApplicationFirewallConfiguration)
+      ? { webApplicationFirewallConfiguration: webApplicationFirewallConfiguration }
+      : {})
+  )
+  zones: zones
+}
+
+resource applicationGateway_lock 'Microsoft.Authorization/locks@2020-05-01' =
+  if (!empty(lock ?? {}) && lock.?kind != 'None') {
+    name: lock.?name ?? 'lock-${name}'
+    properties: {
+      level: lock.?kind ?? ''
+      notes: lock.?kind == 'CanNotDelete'
+        ? 'Cannot delete resource or child resources.'
+        : 'Cannot delete or modify the resource or child resources.'
+    }
+    scope: applicationGateway
+  }
+
+resource applicationGateway_diagnosticSettings 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = [
```

Comment:
Uses outdated schema. Please update (including the UDT, just in case)

You'll find the main difference in the reference of the metrics & logs

Reply:
Updated: https://github.com/Azure/bicep-registry-modules/pull/835/commits/2efe54c70e9079a72c890d8d2e0a8329b042e50b